conans/tools.py

```diff
@@ -1,49 +1,211 @@
+
+# Tools form conans.util
 from conans.util.env_reader import get_env
-# noinspection PyUnresolvedReferences
-from conans.util.files import (_generic_algorithm_sum, load, sha256sum,
-                               sha1sum, md5sum, md5, touch, relative_dirs,
-                               rmdir, mkdir, to_file_bytes)
+from conans.util.files import _generic_algorithm_sum, load, sha256sum, \
+            sha1sum, md5sum, md5, touch, relative_dirs, \
+                              rmdir, mkdir, to_file_bytes, save as files_save, save_append
+
+
+# This global variables are intended to store the configuration of the running Conan application
+_global_output = None
+_global_requester = None
 
 
-def save(path, content, append=False):
-    """
-    This method is duplicated to the conans.util.files, because the
-    intent is to modify the internally used save() methods for unicode
-    encodings
-    """
-    try:
-        os.makedirs(os.path.dirname(path))
-    except:
-        pass
-
-    mode = "ab" if append else "wb"
-    with open(path, mode) as handle:
-        handle.write(to_file_bytes(content))
-
-
-# Global instances
 def set_global_instances(the_output, the_requester):
-    # Assign global variables to needed modules
-    from conans.client.tools import files as _files
-    from conans.client.tools import net as _net
-    from conans.client.tools import oss as _oss
-    from conans.client.tools import system_pm as _system_pm
-    from conans.client.tools import win as _win
-
-    _files._global_output = the_output
-    _oss._global_output = the_output
-    _system_pm._global_output = the_output
-    _win._global_output = the_output
-    _net._global_requester = the_requester
-
-
-# Assign a default, will be overwritten in the Factory of the ConanAPI
-out = ConanOutput(sys.stdout)
-set_global_instances(out, requests)
+    global _global_output
+    global _global_requester
+
+    old_output, old_requester = _global_output, _global_requester
+
+    # TODO: pass here the configuration file, and make the work here (explicit!)
+    _global_output = the_output
+    _global_requester = the_requester
+
+    return old_output, old_requester
+
+
+def get_global_instances():
+    return _global_output, _global_requester
+
+
+# Assign a default, will be overwritten in the factory of the ConanAPI
+set_global_instances(the_output=ConanOutput(sys.stdout, True), the_requester=requests)
+
+
+"""
+From here onwards only currification is expected, no logic
+"""
+
+
+def save(path, content, append=False):
+    # TODO: All this three functions: save, save_append and this one should be merged into one.
+    if append:
+        save_append(path=path, content=content)
+    else:
+        files_save(path=path, content=content, only_if_modified=False)
+
+
+# From conans.client.tools.net
+ftp_download = tools_net.ftp_download
+
+
+def download(*args, **kwargs):
+    return tools_net.download(out=_global_output, requester=_global_requester, *args, **kwargs)
```

Comment:
currify with ``partial``

Reply:
Not using `partial` because it will allow the user to override `out` and `requester` arguments (yes, partial allows to pass already _curryfied_ ones).